content/en/docs/concepts/wasm/index.md

```diff
@@ -0,0 +1,52 @@
+---
+title: WebAssembly (Wasm) Plugins
+description: Describes Istio's WebAssembly Plugin system.
+weight: 25
+keywords: [wasm,webassembly,emscripten,extension,plugin,filter]
+---
+
+WebAssembly is a sandboxing technology which can be used to extend the Istio proxy (Envoy).  Going forward Istio extensions will be written as Envoy proxy plugins using the WebAssembly (Wasm) sandbox API. Mixer will be eliminated from the Istio architecture and all Mixer adapters must be eventually rewritten as proxy plugins.  
+
+WebAseembly sandbox Goal:
+
+- **Efficiency** - An extension adds low latency, CPU, and memory overhead. 
+- **Function** - An extension can enforce policy, collect telemetry, and perform payload mutations.
+- **Isolation** - A programming error or crash in an extension does affect other extensions.
+- **Configuration** - The extensions are configured using an API that is consistent with other Isio APIs. An extension can be configured dynamically.
+- **Operator** - An extension can be canaried and deployed as log-only, fail-open or fail-close.
+- **Extension developer** - The extensions can be written in several programming languages.
+
+This [video talk](https://youtu.be/XdWmm_mtVXI) is an introduction about architecture of WebAssembly integration.
+
+## High-level architecture
+
+Istio extensions have several components:
+
+- **Filter SPI** for building filters
```

Comment:
SPI?

Reply:
That was something that mandar had in his doc.  not sure.... @mandarjog

Reply:
it’s Service Provider Interface.

You implement an SPI and use an API.
If the name bothers you we can just use API everywhere.

Reply:
Added alliteration.